### ISSUE TITLE
Support leading mbox-style From line

### ISSUE THREAD

**Issue by singpolyma** [purebred-mua/purebred-email#38]
Mail messages in various contexts (such as from procmail, etc) often start with an mbox-style From line that is not in header format, such as:

```
From blahblah@blah.tld Wed Jan 15 11:11:37 2020
Headers go here
```

Currently feeding such data into purebred-email causes the parser to fail.  Removing the leading From line allows the mail to be parsed.

Something like this works:

```
mboxFrom = Atto.string (encodeUtf8 $ fromString "From ") *> Atto.skipWhile (\c -> c /= 0x0D && c /= 0x0A) *> MIME.crlf

MIME.parse (mboxFrom *> MIME.message MIME.mime) input
```

**Comment by frasertweedale** [comment]
purebred-email parses the Internet Message format.  mbox is another thing.  I don't mind imposing "strip first line" on users of the library who would parse mbox data.

I am open to pursuasion, but wary of scope creep.

**Comment by singpolyma** [comment]
Somebody claiming to be Fraser Tweedale wrote:
>mbox is another thing.  I don't mind imposing "strip first line" on users 
>of the library who would parse mbox data.

I would agree that actually parsing mbox as a format would be a *major* 
scope creep.  The reason I think this "Ignore leading from line" might be 
worth adding to the parser is that so much mail (even inside a mailserver's 
processing pipeline) ends up with it added, and most unix-y utilities for 
handling mail happily ignore it, so it was quite surprising to me when at 
first the parser failed.

However, as you say, the workaround for me is not super hard so if you don't 
want this to live in purebred-email itself I understand.

**Comment by frasertweedale** [comment]
For reference: https://tools.ietf.org/html/rfc4155.

Reading a bit more, there are a few challenges:

- Escaping lines that already start with `From `.  Escaping lines that already start with whatever the escape sequence for `From ` is.  *Ad infinitum*.

- Determining where messages end.  RFC 5322 is ambivalent (the message ends where the data stops).  MIME messages *might* have a `Content-Length`, or the program that writes the `mbox` might add it.  If the escaping is handled properly you can scan ahead to look for the next `From ...` line, and that will tell you where the message ends.  Then you have to strip the `>From` escaping.  If the mbox contains just a single message, ok no worries, but I'm not interested in releasing a half-arsed implementation as part of purebred-email (or anywhere else).

tl;dr.  It is tractable, and I would consider accepting a PR that supported mbox with multiple messages and handled the escaping properly, but I'm not keen to implement it myself.  So I'll close this.  But we can continue discussion if you (or anyone) wants to tackle it and discuss design / implementation details.
